custom-metrics-stackdriver-adapter/pkg/adapter/provider/translator.go

```diff
@@ -48,24 +51,20 @@ type Translator struct {
 func (t *Translator) GetSDReqForPods(podList *v1.PodList, metricName string) (*stackdriver.ProjectsTimeSeriesListCall, error) {
 	resourceIDs := getResourceIDs(podList)
 	if len(resourceIDs) == 0 {
-		return nil, fmt.Errorf("No objects matching provided selector")
+		return nil, apierr.NewBadRequest("No objects matched to provided selector")
 	}
-	filter, err := t.createFilter(resourceIDs, t.config.Project, t.config.Zone, t.config.Cluster, metricName)
-	if err != nil {
-		return nil, err
-	}
-	return t.createListTimeseriesRequest(filter), nil
+	return t.createListTimeseriesRequest(joinFilters(t.filterForMetric(t.config.MetricsPrefix+"/"+metricName), t.filterForCluster(), t.filterForPods(resourceIDs))), nil
```

Comment:
it's unreadable
could you please extract some params to local vars?

Reply:
Done.